### PR TITLE
Feature/sonar fixes

### PR DESCRIPTION
**Pull request by alexfdezsauco**
- Some bug fixes detected by SONAR

### CODE REVIEW

File: src/Catel.Core/Catel.Core.Shared/Collections/ArrayShim.cs [hunk 1]
```diff
@@ -18,9 +18,9 @@ namespace Catel.Collections
     public static class ArrayShim
     {
 #if ARRAY_EMPTY_MISSING
-        internal static class EmptyArray<T>
```
Comment (GeertvanHorrik) [inline src/Catel.Core/Catel.Core.Shared/Collections/ArrayShim.cs]: Note sure if we can make this change. Shouldn't this class be available to the outside so everyone can use the ArrayShim? Or are there other members that people can still use?

File: src/Catel.Core/Catel.Core.Shared/Logging/Listeners/SeqLogListener.cs [hunk 2]
```diff
@@ -89,46 +85,43 @@ protected override string FormatLogEvent(ILog log, string message, LogEvent logE
...
                 textWriter.Write("]}");
 
                 var message = textWriter.ToString();
 
-                lock (_lock)
+                var baseUri = ServerUrl;
+                if (!string.IsNullOrWhiteSpace(baseUri))
                 {
-                    if (_webClient == null)
+                    using (var webClient = new WebClient { Encoding = Encoding.UTF8 })
```
Comment (GeertvanHorrik) [inline src/Catel.Core/Catel.Core.Shared/Logging/Listeners/SeqLogListener.cs]: I am not sure whether creating a new web client on every request is a good idea (performance wise). I think it's better to let the listener implement IDisposable?

File: src/Catel.Core/Catel.Core.Shared/Logging/Listeners/SeqLogListener.cs [hunk 3]
```diff
@@ -89,46 +85,43 @@ protected override string FormatLogEvent(ILog log, string message, LogEvent logE
...
+                        var webApiUrl = string.Format("{0}{1}", baseUri, BulkUploadResource);
 
-                        _webClient = new WebClient {Encoding = Encoding.UTF8};
-                        _webClient.Headers[HttpRequestHeader.ContentType] = "application/json";
+                        webClient.Headers[HttpRequestHeader.ContentType] = "application/json";
 
                         if (!string.IsNullOrWhiteSpace(ApiKey))
                         {
-                            _webClient.Headers.Add(ApiKeyHeaderName, ApiKey);
+                            webClient.Headers.Add(ApiKeyHeaderName, ApiKey);
```
Comment (GeertvanHorrik) [inline src/Catel.Core/Catel.Core.Shared/Logging/Listeners/SeqLogListener.cs]: We should have a separate method:

CreateWebClient()

this creates a webclient (if needed). Then it adds the default ApiKey as well, so we don't have to do that manually on each call.

Comment (alexfdezsauco) [reply src/Catel.Core/Catel.Core.Shared/Logging/Listeners/SeqLogListener.cs]: Fixed

File: src/Catel.Core/Catel.Core.Shared/Logging/Listeners/SeqLogListener.cs (hunk 2 above)
Comment (alexfdezsauco) [reply src/Catel.Core/Catel.Core.Shared/Logging/Listeners/SeqLogListener.cs]: Fixed

File: src/Catel.Core/Catel.Core.Shared/Collections/ArrayShim.cs (hunk 1 above)
Comment (alexfdezsauco) [reply src/Catel.Core/Catel.Core.Shared/Collections/ArrayShim.cs]: Fixed

**Pull request merged**